src/main/java/lotto/model/LottoResult.java

```diff
@@ -0,0 +1,54 @@
+package lotto.model;
+
+import lotto.model.wrapper.LottoMatchCount;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public enum LottoResult {
+    NONE(LottoMatchCount.of(0), 0L),
+    ONE(LottoMatchCount.of(1), 0L),
+    TWO(LottoMatchCount.of(2), 0L),
+    THREE(LottoMatchCount.of(3), 5000L),
+    FOUR(LottoMatchCount.of(4), 50000L),
+    FIVE(LottoMatchCount.of(5), 1500000L),
+    SIX(LottoMatchCount.of(6), 2000000000L),
+    ;
+
+    private static final Map<LottoMatchCount, LottoResult> LOTTO_RESULTS = Collections.unmodifiableMap(
+            Stream.of(values())
+                    .collect(Collectors.toMap(LottoResult::getLottoMatchCount, Function.identity())));
+
+    private final LottoMatchCount matchCount;
+    private final Long price;
```

Comment:
참조 자료형 대신 기본 자료형을 사용하면 어떨까요? 아래의 글을 참고하세요.
https://jaehun2841.github.io/2019/03/01/effective-jave-item61

```suggestion
    private final long price;
```